src/fft.jl

```diff
@@ -1,63 +1,121 @@
 import FFTW
 
+# returns the lengths of the bounding rectangle in reciprocal space
+# that encloses the sphere of radius Gmax
+function bounding_rectangle(lattice::AbstractMatrix{T}, Gmax; tol=1e-8) where {T}
+    # If |B G| ≤ Gmax, then
+    # |Gi| ≤ |e_i^T B^-1 B G| ≤ |B^-T e_i| Gmax = |A_i| Gmax
+    # with B the reciprocal lattice matrix and A_i the i-th column of the lattice matrix
+    Glims_ = [norm(lattice[:, i]) / 2T(π) * Gmax for i in 1:3]
+
+    # Round up, unless exactly zero (in which case keep it zero in
+    # order to just have one G vector for 1D or 2D systems)
+    Glims = [0, 0, 0]
+    for i = 1:3
+        if Glims_[i] != 0
```

Comment:
Compare Glims_[1] to `tol` as well for consistency? ... not that it matters for practical cases

Reply:
No, we just want to catch exactly zero here